### PR TITLE
feat: migrate to pnpm

### PR DESCRIPTION
**Pull request by SunsetTechuila**

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -3,42 +3,50 @@ on:
...
     name: Push
     runs-on: ubuntu-latest
+    env:
+      TARGET_BRANCH: dist
+    permissions:
+      contents: write
     steps:
-      - name: Check out Git repository
-        uses: actions/checkout@v2.4.0
+      - name: Checkout Project
```
Comment (theRealPadster) [inline .github/workflows/push.yml:16]:
```suggestion
      - name: Check Out Project
```

**Comment by theRealPadster** [comment]
Is this all tested and good to go now?

File: .github/workflows/push.yml (hunk 1 above)
Comment (rxri) [reply .github/workflows/push.yml:16]: checkout is correct though

**Comment by SunsetTechuila** [comment]
don't we need to add all yarn and npm files to gitignore or only allow pnpm?

**Comment by theRealPadster** [comment]
> don't we need to add all yarn and npm files to gitignore or only allow pnpm?

I don't think we need to, but it would be a good idea, yes

**Comment by SunsetTechuila** [comment]
What exactly is a good idea? Add to gitignore or allow only pnpm?

**Comment by theRealPadster** [comment]
Add the files to `.gitignore`, so people don't accidentally commit them. I'm not sure how you'd only allow pnpm.

**Comment by SunsetTechuila** [comment]
packageManager property doesn't force to use it

**Comment by rxri** [comment]
and use `engines` and `engine-strict` option in .npmrc

**Pull request merged**